### PR TITLE
Defer importing ens._normalization to speed up web3 import by 27%

### PR DESCRIPTION
**Pull request by szemate**
### What was wrong?

As of version 7.0.0-beta.2 the `import web3` command takes 375 ms to run*, which is quite long for a library import operation. This affects the usability of command-line tools built around web3.py, where import times are critical.

### How was it fixed?

By deferring the import of `ens._normalization` until the first call of the `ens.utils.normalize_name` function, the execution time of `import web3` is reduced to 275 ms* i.e. by 27%.

### Todo:
- [ ] Add entry to the [release notes](https://github.com/ethereum/web3.py/blob/main/newsfragments/README.md)

---

### Details

Profiling* showed that the `ens._normalization` module performs two lengthy operations at import time:

- Loading `ens/spec/normalization_spec.json` that is 3 Mbytes in size takes 7% of the total web3 import time
- Executing the `_extract_valid_codepoints()` function that performs transformations on the loaded data takes 16% of the total web3 import time

These are very costly operations to perform at every single import, with the sole purpose of initializing the `ens.utils.normalize_name` function.

*: Measurements were taken with `timeit` and `pyinstrument` on an AMD Ryzen 7 machine with M.2 SSD using Python 3.10.12:
   ```py
   >>> timeit.timeit("import web3")
   ```
   and
   ```bash
   $ pyinstrument --show-regex '.*(eth|web3|ens).*' -c 'import web3'
   ```

### CODE REVIEW

**Review by kclowes — APPROVED** [review]
Thanks for tracking this down! I'll add a newsfragment and merge.

**Comment by kclowes** [comment]
Just kidding, looks like I/maintainers can't add to your PR. Once you add a newsfragment I can merge. Thanks!

**Comment by szemate** [comment]
Newsfragment added, thank you!

**Comment by kclowes** [comment]
Thank you!

**Pull request merged**
